jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java

```diff
@@ -1023,6 +1023,9 @@ public void perform(RunContext context) throws InterruptedException {
                     performScenarios(context);
                 }
                 performAfterSteps(context);
+                if (!context.configuration().storyControls().resetStateBeforeScenario()) {
+                    context.configuration().storyControls().currentStoryControls().doResetStateBeforeScenario(true);
+                }
```

Comment:
```suggestion
                if (!context.configuration().storyControls().resetStateBeforeScenario()) {
                    context.configuration().storyControls().resetCurrentStoryControls();
                }
```

and in `StoryControls`:
```java
public void resetCurrentStoryControls() {
        return currentStoryControls.remove();
    }
```